modules/matchRoutes.js

```diff
@@ -44,13 +44,13 @@ function getIndexRoute(route, location, callback) {
 }
 
 function assignParams(params, paramNames, paramValues) {
-  return paramNames.reduceRight(function (params, paramName, index) {
+  return paramNames.reduce(function (params, paramName, index) {
```

Comment:
This was a leftover from a direction I ended up abandoning, but it seems like it simplifies this function a bit to not use `reduceRight`.